# SF311 requests

Opened August 3–9, 2026 · Extracted September 22, 2026


In [1]:
import pandas as pd

requests = pd.read_csv(
    "data/sf311_requests.csv", dtype={"request_id": "string", "district": "Int64"}
)
print("Requests:", len(requests))


Requests: 16876


## By district


In [2]:
original_report = (
    requests.groupby("district", dropna=False)["hours_to_close"]
    .agg(requests="size", with_time="count", median_hours="median")
    .round(2)
    .reset_index()
)
original_report


,district,requests,with_time,median_hours
0,1,1064,1010,5.47
1,2,1347,1236,5.50
2,3,2357,2158,14.14
3,4,697,671,3.61
4,5,1926,1816,12.10
5,6,2139,1995,13.80
6,7,781,721,15.33
7,8,1793,1689,9.40
8,9,2536,2366,6.14
9,10,1232,1139,13.89


## Data checks


In [3]:
audit = requests.copy()
audit["recorded_hours"] = (
    pd.to_datetime(audit["closed_at"]) - pd.to_datetime(audit["opened_at"])
).dt.total_seconds() / 3600


## Street and Sidewalk Cleaning


In [4]:
category_name = "Street and Sidewalk Cleaning"
selected = requests.loc[requests["category"] == category_name].copy()
print("Selected requests:", len(selected))


Selected requests: 6261


In [5]:
draft_report = (
    selected.groupby("district", dropna=False)["hours_to_close"]
    .agg(requests="size", with_time="count", median_hours="median")
    .round(2)
    .reset_index()
)
draft_report


,district,requests,with_time,median_hours
0,1,280,279,4.92
1,2,442,438,3.97
2,3,854,842,11.96
3,4,173,172,3.24
4,5,831,826,8.58
5,6,869,860,13.75
6,7,205,204,14.85
7,8,560,558,3.61
8,9,1152,1142,3.82
9,10,553,535,14.21


## About the data

This report uses 16,876 SF311 requests opened August 3–9, 2026, downloaded on September 22. Each input row is a request; the report groups requests by district. Closure times describe recorded administrative events, not necessarily when the underlying problem was resolved.

See [the data reference](data/README.md) for column definitions, the extraction window and how missing or inconsistent durations are handled.
